framework/src/headers/tcp.rs

```diff
@@ -382,12 +383,27 @@ where
         u16::from_be(self.csum)
     }
 
-    // FIXME: Validate checksum and update checksum
-
     pub fn set_checksum(&mut self, csum: u16) {
         self.csum = u16::to_be(csum)
     }
 
+    /// Update Checksum w/ tcp_segment_length (from <Packet>) and src and dst on
+    /// header.
+    /// TODO: Validate Checksum?
+    pub fn update_checksum(&mut self, segment_length: usize, src: Ipv6Addr, dst: Ipv6Addr) {
+        self.set_checksum(0);
+
+        let mut sum = segment_length as u32
+            + src.segments().iter().map(|x| *x as u32).sum::<u32>()
```

Comment:
Nice! This `segments` function seems made for the checksum.

Reply:
yep, the only key was figuring out the right casting needs.